test/test_cartesian_graph.py

```diff
@@ -157,13 +157,13 @@ def test_to_torch_and_backprop():
         loss.backward()
 
         optimizer.step()
-    assert pytest.approx(float(loss.detach()) == 0.0)
+    assert float(loss.detach()) == pytest.approx(0.0)
 
     x = [3.0]
     x_torch = torch.Tensor(x).view(1, 1)
-    assert pytest.approx(c(x_torch)[0].detach().numpy() != graph(x))
+    assert c(x_torch)[0].detach().numpy() != pytest.approx(graph(x))
     graph.update_parameters_from_torch_class(c)
-    assert pytest.approx(c(x_torch)[0].detach().numpy() == graph(x))
+    assert c(x_torch)[0][0].detach().numpy() == pytest.approx(graph(x)[0].detach().numpy())
```

Comment:
why the extra zero here? shouldn't it work just like two lines above?

Reply:
See below, your suggested approach does not work on my machine, so I need to transform the right hand side. To be able to call `detach().numpy()` I need to do `[0]` first, because it is a `list`.

When I do that, I need to take the extra `[0]` on the left hand side.

Reply:
ah interesting, this is a bug! the `update_parameters_from_torch_class` changes the output of the corresponding node to a torch parameter (https://github.com/jakobj/python-gp/blob/master/gp/cartesian_graph.py#L290). the line in `cartesian_graph` should read `n._output = eval(f"torch_cls._p{n._idx}[0].item()")` instead to extract the value and avoid setting the node output to a torch parameter. could you fix that quickly in this PR so you can use the suggested comparison without the detach etc?

Reply:
Okay, done in #74 . Note that #74 now has to be merged first to make CI pass on this PR.